File: java/src/test/java/challenges/aoc2024/day5/model/PageOrderingRuleTest.java
package challenges.aoc2024.day5.model;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

class PageOrderingRuleTest {

    @Test
    void shouldBuildPageOrderingRuleObjectFromString() {
        // When
        final PageOrderingRule pageOrderingRule = PageOrderingRule.of("23|47");

        // Then
        assertEquals(ImmutablePageOrderingRule.class, pageOrderingRule.getClass());
        assertEquals(23, pageOrderingRule.getBeforeNumber());
        assertEquals(47, pageOrderingRule.getAfterNumber());
    }
}

File: scala/src/test/scala/challenges/aoc2024/day5/model/PageOrderingRuleTest.scala
package challenges.aoc2024.day5.model

import org.scalatest.funsuite.AnyFunSuite

class PageOrderingRuleTest extends AnyFunSuite {

  test("should build PageOrderingRule object from string") {
    // When
    val pageOrderingRule: PageOrderingRule = PageOrderingRule.of("23|47")

    // Then
    assert(pageOrderingRule.isInstanceOf[PageOrderingRule])
    assert(23 == pageOrderingRule.beforeNumber)
    assert(47 == pageOrderingRule.afterNumber)
  }
}
